src/www/app/util.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { OperationTimedOut } from '../model/errors';
+import { timeoutPromise } from './util';
+
+const PROMISE_RESOLVED = 1;
+
+describe('timeoutPromise', () => {
+  it('Executes successful promise', () => {
+    timeoutPromise(Promise.resolve(PROMISE_RESOLVED), 100, 'Test Promise')
+        .catch((err) => {
+          fail(`Successful promise was timed out when it should have resolved`);
+        })
+        .then((value) => {
+          expect(value).toEqual(PROMISE_RESOLVED);
+        });
+    const promiseWithTime = new Promise((resolve, _) => {
+      setTimeout(() => {
+        resolve(PROMISE_RESOLVED);
+      }, 50);
+    });
+    timeoutPromise(promiseWithTime, 100, 'Test Promise')
+        .catch((err) => {
+          fail(`Successful timed promise was timed out when it should have resolved`);
+        })
+        .then((value) => {
+          expect(value).toEqual(PROMISE_RESOLVED);
+        });
+  });
+
+  it('Executes failed promise', () => {
+    timeoutPromise(Promise.reject('reason'), 100, 'Test Promise').catch((err) => {
+      expect(err).not.toEqual(jasmine.any(OperationTimedOut));
+    });
+    const promiseWithTime = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(PROMISE_RESOLVED);
+      }, 50);
+    });
+    timeoutPromise(promiseWithTime, 100, 'Test Promise').catch((err) => {
```

Comment:
Why do you expect the promise to fail? The promise should have been resolved, since it's under the timeout.

Reply:
Because that's supposed to be a failing promise